### PR TITLE
Add Bluepy==2.3.0, morphio==3.1.1, bluepysnap==0.12.0

### PR DESCRIPTION
**Pull request by tomdele**

### CODE REVIEW

**Comment by matz-e** [comment]
This is weird, seems like the full CI wasn't triggered.

**Comment by tomdele** [comment]
Hi @matz-e,
Thanks for the information.
 
Does it mean the failure is a system failure or a package problem ?

**Comment by matz-e** [comment]
There's an issue concretizing `py-morphology-repair-workflow`…

I meant there was a button here on the UI saying "approve and run CI" to get the Github stuff running. Seems weird.

**Comment by GianlucaFicarelli** [comment]
> There's an issue concretizing `py-morphology-repair-workflow`…

This is unexpected for me, for `py-morphology-repair-workflow` there is this other PR https://github.com/BlueBrain/spack/pull/1099 that I was planning to rebase after this PR.

I think that the spack package for `py-morphology-repair-workflow` is not working anyway, maybe in this PR it can be just fixed to pass the CI?

**Comment by tomdele** [comment]
> I think that the spack package for py-morphology-repair-workflow is not working anyway, maybe in this PR it can be just fixed to pass the CI?

I am fine with this

**Comment by tomdele** [comment]
I am struggling to find the sub-package calling a pinned version of morphio and causing the problem in sonata-network-reduction. Is there a way of exploring the tree ?

**Comment by matz-e** [comment]
> I am struggling to find the sub-package calling a pinned version of morphio and causing the problem in sonata-network-reduction. Is there a way of exploring the tree ?

I normally follow [this procedure](https://github.com/BlueBrain/spack/blob/develop/deploy/docs/debug_bb5.md), and often refer back to previous Jenkins logs for a full spec dump.

File: var/spack/repos/builtin/packages/morphio/package.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Morphio(CMakePackage):
     git      = "https://github.com/BlueBrain/MorphIO.git"
 
     version('develop', git=url, submodules=True)
+    version('3.1.1', tag='v3.1.1', submodules=True)
     version('3.0.2', tag='v3.0.2', submodules=True)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/morphio/package.py:18]: Please update these versions in `py-morphio`, too :)

**Review by matz-e — CHANGES_REQUESTED** [review]
Should have paid more attention. The `morphio` package is used as a C++ dependency, but we also need updates to `py-morphio`!

File: var/spack/repos/builtin/packages/morphio/package.py (hunk 1 above)
Comment (tomdele) [reply var/spack/repos/builtin/packages/morphio/package.py:18]: my god ... I am so dumb

Comment (matz-e) [reply var/spack/repos/builtin/packages/morphio/package.py:18]: Took me quite a while to notice this myself. After grepping for `depends_on.*py-morphio` not giving anything useful, that was the only place left to look.

Comment (tomdele) [reply var/spack/repos/builtin/packages/morphio/package.py:18]: Yeah I have been grepping the py-morphio and all the possible dependencies containing it. I totally missed/forgot the normal morphio. Happens :)

**Comment by matz-e** [comment]
Retest this please

Seems like we're running in some concurrency issues :(

**Comment by tomdele** [comment]
From https://bbpcode.epfl.ch/ci/job/hpc.spack-deployment.pull-requests/1908/console
I am not 100% sure this is not a system problem.

**Comment by tomdele** [comment]
Retest this please

**Comment by matz-e** [comment]
> From https://bbpcode.epfl.ch/ci/job/hpc.spack-deployment.pull-requests/1908/console
> I am not 100% sure this is not a system problem.

I can't see anything in these logs if it isn't through the Blue Ocean interface :)

Some stuff builds, otherwise still weird locking stuff probably.

Retest this please.

**Comment by tomdele** [comment]
the infamous : 
```
16:15:29  Fetching tags only, you probably meant:
16:15:29    git fetch --tags
16:15:29  
16:15:29  gzip: stdin: unexpected end of file
16:15:29  /usr/bin/tar: Unexpected EOF in archive
16:15:29  /usr/bin/tar: Unexpected EOF in archive
```

It looks like an Heisenbug. The fails are different between runs...

**Comment by matz-e** [comment]
> It looks like an Eisenbug. The fails are different between runs...

Heisenbug? Yeah… sometimes concurrency is pretty… bad.

Retest this please.

**Comment by tomdele** [comment]
Ok this one makes sense. I will correct

File: var/spack/repos/builtin/packages/py-morphology-repair-workflow/package.py [hunk 2]
```diff
@@ -19,8 +19,8 @@ class PyMorphologyRepairWorkflow(PythonPackage):
...
-    depends_on('py-neuroc', type='run')
-    depends_on('py-morph-tool', type='run')
-    depends_on('py-neuror', type='run')
-    depends_on('py-more-itertools', type='run')
-    depends_on('py-morph-validator', type='run')
+    # depends_on('py-neuroc', type='run')
+    # depends_on('py-morph-tool', type='run')
+    # depends_on('py-neuror', type='run')
+    # depends_on('py-more-itertools', type='run')
+    # depends_on('py-morph-validator', type='run')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-morphology-repair-workflow/package.py:26]: can we add these again? or is this a future PR?

**Review by matz-e — APPROVED** [review]
final question, but otherwise I'm happy

File: var/spack/repos/builtin/packages/py-morphology-repair-workflow/package.py (hunk 2 above)
Comment (tomdele) [reply var/spack/repos/builtin/packages/py-morphology-repair-workflow/package.py:26]: This is for a future PR by @GianlucaFicarelli

**Comment by tomdele** [comment]
Thanks a lot @matz-e for the support again !

File: var/spack/repos/builtin/packages/py-morphology-repair-workflow/package.py (hunk 2 above)
Comment (tomdele) [reply var/spack/repos/builtin/packages/py-morphology-repair-workflow/package.py:26]: should be done quickly

Comment (tomdele) [reply var/spack/repos/builtin/packages/py-morphology-repair-workflow/package.py:26]: Also the current module is not working anyway from what I understood

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-morphology-repair-workflow/package.py:26]: Ok, thank you for clarifying! :)

**Pull request merged**